app/controllers/api/messaging.php

```diff
@@ -0,0 +1,1277 @@
+
+        if ($provider->isEmpty()) {
+            throw new Exception(Exception::PROVIDER_NOT_FOUND);
+        }
+
+        $dbForProject->deleteDocument('providers', $provider->getId());
+
+        $response->noContent();
+    });
+
+App::post('/v1/messaging/messages/email')
+    ->desc('Send an email.')
+    ->groups(['api', 'messaging'])
+    ->label('audits.event', 'messages.create')
+    ->label('audits.resource', 'messages/{response.$id}')
+    ->label('scope', 'messages.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN, APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'messaging')
+    ->label('sdk.method', 'sendEmail')
+    ->label('sdk.description', '/docs/references/messaging/send-email.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_MESSAGE)
+    ->param('messageId', '', new CustomId(), 'Message ID. Choose a custom ID or generate a random ID with `ID.unique()`. Valid chars are a-z, A-Z, 0-9, period, hyphen, and underscore. Can\'t start with a special char. Max length is 36 chars.')
+    ->param('providerId', '', new UID(), 'Email Provider ID.')
+    ->param('to', [], new ArrayList(new Text(Database::LENGTH_KEY)), 'List of Topic IDs or List of User IDs or List of Target IDs.')
+    ->param('subject', '', new Text(998), 'Email Subject.')
+    ->param('description', '', new Text(256), 'Description for Message.', true)
+    ->param('content', '', new Text(64230), 'Email Content.')
+    ->param('html', false, new Boolean(), 'Is content of type HTML', true)
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('messaging')
+    ->inject('response')
+    ->action(function (string $messageId, string $providerId, array $to, string $subject, string $description, string $content, string $html, Database $dbForProject, Document $project, Messaging $messaging, Response $response) {
+        $messageId = $messageId == 'unique()' ? ID::unique() : $messageId;
+
+        $provider = $dbForProject->getDocument('providers', $providerId);
+
+        if ($provider->isEmpty()) {
+            throw new Exception(Exception::PROVIDER_NOT_FOUND);
+        }
+
+        $message = $dbForProject->createDocument('messages', new Document([
+            '$id' => $messageId,
+            'providerId' => $provider->getId(),
+            'providerInternalId' => $provider->getInternalId(),
+            'to' => $to,
+            'data' => [
+                'subject' => $subject,
+                'content' => $content,
+                'html' => $html,
+                'description' => $description,
+            ],
+            'status' => 'processing',
+            'search' => $messageId . ' ' . $description . ' ' . $subject,
+        ]));
+
+        $messaging
+            ->setMessageId($message->getId())
+            ->setProject($project)
+            ->trigger();
```

Comment:
If the status is draft, we will need to skip this trigger. We'll also need to add an update message endpoint which is missed in the RFC

Reply:
done this, added update message endpoint as well